suite-native/device/src/hooks/useWipeDevice.tsx

```diff
@@ -1,47 +1,58 @@
-import { useCallback, useState } from 'react';
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { useNavigation } from '@react-navigation/native';
+import { isFulfilled } from '@reduxjs/toolkit';
+import { useSetAtom } from 'jotai';
 
-import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { selectSelectedDevice, wipeDeviceThunk } from '@suite-common/wallet-core';
 import { requestPrioritizedDeviceAccess } from '@suite-native/device-mutex';
 import {
-    DeviceOnboardingStackRoutes,
-    RootStackParamList,
+    DeviceSettingsStackRoutes,
     RootStackRoutes,
-    StackNavigationProps,
+    WipeDeviceStackRoutes,
 } from '@suite-native/navigation';
-import TrezorConnect from '@trezor/connect';
 
-type NavigationProp = StackNavigationProps<RootStackParamList, RootStackRoutes.BackupFailedModal>;
+import { wasDeviceOnboardingCancelledAtom } from '../deviceAtoms';
 
 export const useWipeDevice = () => {
-    const [isWipeInProgress, setIsWipeInProgress] = useState(false);
-    const navigation = useNavigation<NavigationProp>();
+    const dispatch = useDispatch();
+    const navigation = useNavigation();
     const device = useSelector(selectSelectedDevice);
+    const setWasDeviceOnboardingCancelled = useSetAtom(wasDeviceOnboardingCancelledAtom);
 
-    const wipeDevice = useCallback(async () => {
-        setIsWipeInProgress(true);
+    const wipeDevice = async () => {
         if (!device) return;
 
+        // After wipe, device gets changed and reconnected. That would trigger redirect to device onboarding which is
+        // not wanted here. We want to treat it differently since it was wiped so user goes to onboarding through homescreen.
+        setWasDeviceOnboardingCancelled(true);
+
+        // @ts-expect-error
```

Comment:
nit: it not there a way how to type `useNavigation` rather than using ignores?

Reply:
As mentioned on slack, I'm trying to figure out a way how to type it, so this one is still in progress. However it's tricky with this one. We might need another much complicated utility type for this, unless I'm missing something.